wordpress-mobile/WordPress-FluxC-Android
# Issue/1556 fetch shipping labels for order

Fixes #1556 by adding support to fetch shipping labels for an order. 

#### Changes
- Even though, shipping labels are part of the orders feature, the endpoints differ so I chose to create new classes/rest clients for the implementation. Also nice to use coroutines 😄
- Adds new endpoint `/wc/v1/connect/label/<orderId>`.  Since this is a new implementation, added all subsequent logic under the `shippinglabels` package.
- Added new `WCShippingLabelModel` model class and a migration script to create table.
- Added new classes, stores and mapper classes to fetch, store, update, delete shipping labels for a site.
- Added unit tests.
- Added a new page to the example app to fetch shipping labels for an order.

#### Screenshots
<img width="300" src="https://user-images.githubusercontent.com/22608780/81919508-922a6700-95f5-11ea-87ad-7e9b1e18ae5f.png"> . <img width="300" src="https://user-images.githubusercontent.com/22608780/81919516-93f42a80-95f5-11ea-8d81-7c362f026af6.png"> 

#### Testing
- Since this PR adds a migration script, it would be nice to test updating the app.
- Run `WCShippingLabelStoreTest` and `WCShippingLabelSqlUtilsTest`.
- In the example app, click on `Woo` -> `Shipping Labels` -> `Select Site` -> `Fetch Shipping Labels`.
  - Enter an invalid order number and verify that an error message is displayed. 
  - Enter a valid order number without any shipping labels and verify that the list is empty, while the request is successful. 
  - Enter a valid order number with shipping labels and verify that the shipping labels are fetched successfully. 
   - Try deleting or refunding a shipping label for the order and verify it is updated in the app.
  - Try fetching shipping labels for multiple orders and verify the correct list is fetched and displayed.
